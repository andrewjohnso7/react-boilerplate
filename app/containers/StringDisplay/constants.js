export const FETCH_STRINGS = 'FETCH_STRINGS';
export const API_SUCCESS = 'API_SUCCESS';
export const API_FAILURE = 'API_FAILURE';
