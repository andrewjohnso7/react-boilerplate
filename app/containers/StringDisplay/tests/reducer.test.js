import fetchStringsReducer from '../reducer';
import { FETCH_STRINGS, FETCH_SUCCESS } from '../constants';

describe('fetchStringsReducer', () => {
  it('returns the initial state when defaulting', () => {
    expect(fetchStringsReducer(undefined, {})).toMatchSnapshot();
  });
  it('handles the FETCH_STRINGS action', () => {
    expect(fetchStringsReducer({}, FETCH_STRINGS)).toMatchSnapshot();
  });
  it('handles the FETCH_SUCCESS action', () => {
    expect(fetchStringsReducer({}, FETCH_SUCCESS)).toMatchSnapshot();
  });
});
