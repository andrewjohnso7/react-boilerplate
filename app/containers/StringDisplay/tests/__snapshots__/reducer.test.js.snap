// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchStringsReducer handles the FETCH_STRINGS action 1`] = `Object {}`;

exports[`fetchStringsReducer handles the FETCH_SUCCESS action 1`] = `Object {}`;

exports[`fetchStringsReducer returns the initial state 1`] = `
Object {
  "list": Array [
    Object {
      "body": "Aang",
      "id": 5,
    },
    Object {
      "body": "Katara",
      "id": 4,
    },
    Object {
      "body": "Toph",
      "id": 3,
    },
    Object {
      "body": "Sokka",
      "id": 2,
    },
    Object {
      "body": "Appa",
      "id": 1,
    },
  ],
}
`;
